app/main.py

```diff
@@ -1,2 +1,82 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.loadfactor = 2 / 3
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+        self.keys = []
+
+    def __add_hash_node(self, key: Any, value: Any) -> None:
+        hash_num = hash(key)
+        slot = hash_num % self.capacity
+        if self.hash_table[slot] is None:
+            self.hash_table[slot] = (key, hash_num, value)
+        else:
+            while self.hash_table[slot] is not None:
+                slot = slot + 1 if slot < self.capacity - 1 else 0
+            self.hash_table[slot] = (key, hash_num, value)
+
+    def __resize_hash_table(self) -> None:
+        self.hash_table += [None] * self.capacity
+        self.capacity = self.capacity * 2
+        for index, hash_node in enumerate(self.hash_table):
+            if hash_node is not None:
+                key, hash_num, value = hash_node
+                self.hash_table[index] = None
+                self.__add_hash_node(key, value)
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_num = hash(key)
+        try:
+            self.__getitem__(key)
+        except KeyError:
+            self.size += 1
+            if self.size > self.capacity * self.loadfactor:
```

Comment:
create a property for `self.size > self.capacity * self.loadfactor`. It will increase readability 
```
@property
def is_need_resize(self):
    return self.size > self.capacity * self.loadfactor
```